static/app/views/dashboards/manage/dashboardList.tsx

```diff
@@ -160,59 +182,46 @@ function DashboardList({
   }
 
   function renderDashboardGrid() {
-    if (!dashboards?.length) {
+    if (!dashboards?.length && !isLoading) {
       return (
         <EmptyStateWarning>
           <p>{t('Sorry, no Dashboards match your filters.')}</p>
         </EmptyStateWarning>
       );
     }
-    return <DashboardGrid>{renderMiniDashboards()}</DashboardGrid>;
-  }
 
-  return (
-    <Fragment>
-      {renderDashboardGrid()}
-      <PaginationRow
-        pageLinks={pageLinks}
-        onCursor={(cursor, path, query, direction) => {
-          const offset = Number(cursor?.split?.(':')?.[1] ?? 0);
+    const numDashboards =
+      rows * columns !== currentDashboards?.length
+        ? currentDashboards?.length
+          ? currentDashboards.length
+          : 0
+        : dashboards?.length
+          ? dashboards.length
+          : 0;
```

Comment:
👀 is it possible to express this ternary in another way? It's really complicated! Maybe via array slicing?